src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -81,7 +81,7 @@ public override Task DocumentResetAsync(Document document, CancellationToken can
                 // we remove whatever information we used to have on document open/close and re-calculate diagnostics
                 // we had to do this since some diagnostic analyzer changes its behavior based on whether the document is opened or not.
                 // so we can't use cached information.
-                ClearDocumentStates(document, _stateManger.GetStateSets(document.Project), cancellationToken);
+                ClearDocumentStates(document, _stateManger.GetStateSets(document.Project), raiseEvent: false, cancellationToken: cancellationToken);
```

Comment:
this is making our behavior same as before. previously, we didn't raise event for document reset, we just wiped out caches for them. when I did refactoring before around state management, I merged some code and now we raise event even for resetting. 

one of impact visible due to that is when re-analysis is requested, diagnostics will flash. now it shouldn't